components/operator/internal/resources/stacks/controller.go

```diff
@@ -46,6 +151,16 @@ func Reconcile(ctx Context, stack *v1beta1.Stack) error {
 		return err
 	}
 
+	err = hasSomeModule(ctx, stack)
```

Comment:
This will return an error if the stack does not have any module.
From a business point of view, it's wrong. A stack with no modules is just ready.

Reply:
You don't really need this, the `areDependentReady` is enough.

Reply:
And why not?
A stack is just an objet, you can create a stack without any module, it is a fact.
And this code will leverage an error, where there are no errors.